vertica/datadog_checks/vertica/vertica.py

```diff
@@ -92,6 +92,37 @@ def __init__(self, name, init_config, instances):
         # Cache database results for re-use among disparate functions
         self._view = defaultdict(list)
 
+        self._tables = {
+            "licenses": self.query_licenses,
+            "license_audits": self.query_license_audits,
+            "system": self.query_system,
+            "nodes": self.query_nodes,
+            "projections": self.query_projections,
+            "projection_storage": self.query_projection_storage,
+            "storage_containers": self.query_storage_containers,
+            "host_resources": self.query_host_resources,
+            "query_metrics": self.query_query_metrics,
+            "resource_pool_status": self.query_resource_pool_status,
+            "disk_storage": self.query_disk_storage,
+            "resource_usage": self.query_resource_usage,
+            "version": self.query_version,
+            "custom": self.query_custom,
+        }
+
+        include_tables = self.instance.get("include_tables") or list(self._tables)
+        exclude_tables = self.instance.get("exclude_tables") or list()
+        desired_tables = set(include_tables) - set(exclude_tables)
+
+        # License query needs to be run before getting system
+        if 'system' in desired_tables:
+            desired_tables.add('licenses')
+
+        self._desired_tables = []
+
+        for table_name in self._tables:
+            if table_name in desired_tables:
+                self._desired_tables.append(table_name)
```

Comment:
minor nit: Instead of checking through the entire dictionary, we can just go through the desired tables list.

```suggestion
        for table_name in desired_tables:
            if table_name in self._tables:
                self._desired_tables.append(table_name)
            else:
                self.log.debug("Table name `%s` is not valid, ignoring table", table_name)
```

Let's add a debug log if any tables listed are invalid

Reply:
Ended up throwing a ConfigurationError for validation (from the newer comment below), so I didn't add the debug log anymore.